### PR TITLE
Update contributing md default nix

### PR DESCRIPTION
**Pull request by refaelsh**
This PR is a direct follow-up of the comment: https://github.com/nix-community/nixvim/pull/2115#issuecomment-2318884967.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -75,6 +75,9 @@ In either case, you don't need to bother implementing this part. It is done auto
 > `settings` has a "freeform" type `attrsOf anything`, meaning it can be configured with _anything_.
 > Declaring `settingsOptions` is therefore optional and just adds type-checking and documentation.
 
+#### Add to lib/default.nix
+As a final step, please add your plugin to `lib/default.nix`. Please maintain the file hierarchy.
```
Comment (MattSturgeon) [inline CONTRIBUTING.md:79]: `lib/default.nix` -> `plugins/default.nix`

**Comment by MattSturgeon** [comment]
Also, I've been thinking about adding a `plugins/by-name` that is automatically imported using `builtins.readDir`. That might simplify the process of adding new plugins a little?

File: CONTRIBUTING.md (hunk 1 above)
Comment (refaelsh) [reply CONTRIBUTING.md]: Done.

**Comment by khaneliman** [comment]
> Also (out of scope for this PR), I've been thinking about adding a `plugins/by-name` that is automatically imported using `builtins.readDir`. That might simplify the process of adding new plugins a little?

I was thinking about that when this was being added. We'll have to just make sure to update the contributing document with that change.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -75,6 +75,9 @@ In either case, you don't need to bother implementing this part. It is done auto
 > `settings` has a "freeform" type `attrsOf anything`, meaning it can be configured with _anything_.
 > Declaring `settingsOptions` is therefore optional and just adds type-checking and documentation.
 
+#### Add to plugins/default.nix
+As a final step, please add your plugin to plugins/default.nix`. Please maintain the file hierarchy.
```
Comment (MattSturgeon) [inline CONTRIBUTING.md:79]:
```suggestion
As a final step, please add your plugin to `plugins/default.nix`. Please maintain the file hierarchy.
```

Just missing a `` ` `` 🙃

Comment (refaelsh) [reply CONTRIBUTING.md]: > Just missing a `` ` `` 🙃

Silly me :-)

**Review by MattSturgeon — APPROVED** [review]
Awesome, thanks for taking the time to add this!

**Comment by refaelsh** [comment]
> Awesome, thanks for taking the time to add this!

I had fun :-)
And thank you for reviewing it.

**Pull request merged**